sdk/resources/Azure.ResourceManager.Resources/src/Generated/Deployment.cs

```diff
@@ -338,58 +338,57 @@ public virtual Response<DeploymentExportResult> ExportTemplate(CancellationToken
         }
 
         /// <summary>
-        /// Returns changes that will be made by the deployment if executed at the scope of the tenant group.
-        /// Request Path: /providers/Microsoft.Resources/deployments/{deploymentName}/whatIf
-        /// Operation Id: Deployments_WhatIfAtTenantScope
-        /// Request Path: /providers/Microsoft.Management/managementGroups/{groupId}/providers/Microsoft.Resources/deployments/{deploymentName}/whatIf
-        /// Operation Id: Deployments_WhatIfAtManagementGroupScope
+        /// Returns changes that will be made by the deployment if executed at the scope of the subscription.
         /// Request Path: /subscriptions/{subscriptionId}/providers/Microsoft.Resources/deployments/{deploymentName}/whatIf
         /// Operation Id: Deployments_WhatIfAtSubscriptionScope
         /// Request Path: /subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Resources/deployments/{deploymentName}/whatIf
         /// Operation Id: Deployments_WhatIf
+        /// Request Path: /providers/Microsoft.Resources/deployments/{deploymentName}/whatIf
+        /// Operation Id: Deployments_WhatIfAtTenantScope
+        /// Request Path: /providers/Microsoft.Management/managementGroups/{groupId}/providers/Microsoft.Resources/deployments/{deploymentName}/whatIf
+        /// Operation Id: Deployments_WhatIfAtManagementGroupScope
         /// </summary>
         /// <param name="waitForCompletion"> Waits for the completion of the long running operations. </param>
-        /// <param name="location"> The location to store the deployment data. </param>
         /// <param name="properties"> The deployment properties. </param>
+        /// <param name="location"> The location to store the deployment data. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="location"/> or <paramref name="properties"/> is null. </exception>
-        public async virtual Task<ArmOperation<WhatIfOperationResult>> WhatIfAsync(bool waitForCompletion, string location, DeploymentWhatIfProperties properties, CancellationToken cancellationToken = default)
+        /// <exception cref="ArgumentNullException"> <paramref name="properties"/> is null. </exception>
+        public async virtual Task<ArmOperation<WhatIfOperationResult>> WhatIfAsync(bool waitForCompletion, DeploymentWhatIfProperties properties, string location = null, CancellationToken cancellationToken = default)
```

Comment:
The parameter `location` is not allowed at the resource group and subscription scope, so I think it's reasonable here to set its default value. The null check of this value will happen in the rest operation at the tenant and management group scope.